websites/T/TikTok/presence.ts

```diff
@@ -41,25 +41,52 @@ presence.on("UpdateData", async () => {
 		presenceData.state = state;
 	} else if (page.startsWith("@")) {
 		//User
-
+		if (document.querySelector("#BreadcrumbList")) {
+			presenceData.state = JSON.parse(
+				document.querySelector("#BreadcrumbList").innerHTML
+			).itemListElement[1].item.name.replace("| TikTok", "");
+		} else {
+			const id2 = document
+				.querySelector(
+					"#app > div.tiktok-ywuvyb-DivBodyContainer.etsvyce0 > div.tiktok-w4ewjk-DivShareLayoutV2.epodnl40 > div > div.tiktok-1g04lal-DivShareLayoutHeader-StyledDivShareLayoutHeaderV2.epodnl42 > div.tiktok-1gk89rh-DivShareInfo.e198b7gd2 > div.tiktok-1hdrv89-DivShareTitleContainer.e198b7gd3 > h2"
+				)
+				.textContent.trim();
+			presenceData.state = `@(${id2}) ${
+				JSON.parse(
+					document
+						.querySelector("#sigi-persisted-data")
+						.innerHTML.slice(0, -486)
+						.replace("window['SIGI_STATE']=", "")
+				).UserModule.users[id2].nickname
+			}`;
+		}
 		if (pageType === "video") {
 			//Video
-
-			const video = document.querySelector<HTMLVideoElement>(".video-player");
+			const time = document
+				.querySelector('[class*="-DivSeekBarTimeContainer"]')
+				.innerHTML.split("/");
 
 			delete presenceData.startTimestamp;
 			presenceData.details =
 				document.querySelector(".video-meta-title:nth-child(1)")
 					?.firstElementChild?.textContent ??
 				document.querySelector(".tt-video-meta-caption")?.firstElementChild
 					?.textContent;
```

Comment:
These don't exist anymore, so either change it to use the `tiktok.viewing` string or use the element below.
```ts
document.querySelector("[data-e2e=browse-video-desc]");
```